modules/nf-core/untar/main.nf

```diff
@@ -20,24 +20,24 @@ process UNTAR {
     script:
     def args  = task.ext.args ?: ''
     def args2 = task.ext.args2 ?: ''
-    untar     = archive.toString() - '.tar.gz'
+    untar     = "${archive.simpleName}"
```

Comment:
`simpleName` would return just `hello`. It strips out everything after the dot, cf https://www.nextflow.io/docs/latest/script.html?highlight=getname#check-file-attributes

This `untar` variable is in fact the name of the output directory, so I think it'd be fine setting it to the usual prefix as per the nf-core pipeline.
Otherwise, you could try something like this:
```suggestion
    untar     = archive.extension == "tar" ? "${archive.baseName}" : "${archive.baseName.baseName}"
```

Reply:
Yeah, the issue is that there's no way of stripping only the extensions we want- since we could have .tgz, tar.gz etc. So I chose to be brutal. I guess it _could_ create issues if people were uncompressing multiple things like 'foo.one.tar.gz', 'foo.two.tar.gz' and ending up with multiple input folders named 'foo' in agglomerative processes. 

`$prefix` is a good call- I've done that (default is the simpleName).

Reply:
Prefix is a good call. I've set the nextflow.config for the tests to override it to preserve existing behaviour (and any existing tools that rely on the directory name for some reason (e.g. gathering unzipped dirs in a process) can do the same.

Reply:
Here's my revised proposal:

- Prefix if specified (which I've done in the tests to match previous behaviour)
- else if meta ID is set use that (per normal prefix behaviour)
- else if prefix is unset, generate a directory based on the archive, by removing extensions with or without an additional .gz extension (via regex).

This will preserve existing behaviour (foo.tar.gz -> foo) where there is no prefix or meta id (which seems to be the bulk of existing usage) and allow use of the conventional prefix.